linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -410,6 +410,10 @@
   delay: 10
   when: uri_hostname == 'localhost'
 
+- set_fact: 
+    static_ip: "{{extract_ip_address_result.results.0['stdout']|omit_filter(omit)}}"
```

Comment:
`0` takes the first ip found and later you assign that IP to all networks with "static IP", which is not right.